tests/run/cpp_iterators.pyx

```diff
@@ -98,6 +103,36 @@ def test_custom():
     finally:
         del iter
 
+def test_custom2():
+    """
+    (deref(iter) instead of iter[0])
+    >>> test_custom()
+    [1.0, 2.0, 3.0]
+    """
+    cdef double* values = [1, 2, 3]
+    cdef DoublePointerIter* iter
+    try:
+        iter = new DoublePointerIter(values, 3)
+        return [x for x in deref(iter)]
+    finally:
+        del iter
+
+def test_custom_genexp():
+    """
+    >>> test_custom_genexp()
+    [1.0, 2.0, 3.0]
+    """
+    def to_list(g):  # function to hide the intent to avoid inlined-generator expression optimization
+        return list(g)
+    cdef double* values = [1, 2, 3]
+    cdef DoublePointerIterCanMakeEmpty* iter
+    try:
+        iter = new DoublePointerIterCanMakeEmpty(values, 3)
```

Comment:
```suggestion
    cdef DoublePointerIterDefaultConstructable* iter
    try:
        iter = new DoublePointerIterDefaultConstructable(values, 3)
```

Reply:
Nit: [constructible](https://www.thefreedictionary.com/constructible) is [arguably](https://wikidiff.com/content/constructable-vs-constructible-whats-difference-0) more correct than constructable.

Reply:
Ah, not only that, but now it has diverged from the header, which defines `DoublePointerIterDefaultConstructible`.

Reply:
I think I copy-pasted your spelling for the class but retyped it myself here... thanks